### PR TITLE
Add :prompt-accept --save for boolean prompts

### LINKED ISSUE #832: Permission answers not saved?
Might just be user error, but it seems that on close, sites for which I've enabled desktop notifications (like slack or gmail) forget that they're allowed to make desktop notifications: when I open back up and reopen those sites, they ask for notification permissions again.

### PR DESCRIPTION
**Pull request by The-Compiler**
Fixes #832

Missing tests as I'm running out of battery, a review of what's here already would be good!

### CODE REVIEW

File: qutebrowser/mainwindow/prompt.py [hunk 1]
```diff
@@ -860,7 +872,14 @@ def __init__(self, question, parent=None):
         self._init_texts(question)
         self._init_key_label()
 
-    def accept(self, value=None):
+    def _check_save_support(self, save):
+        if self.question.option is None:
```
Comment (toofar) [inline qutebrowser/mainwindow/prompt.py]: Should this be `if save and self.question.option is None:`?

**Comment by jgkamat** [comment]
Overall this looks good to me, but I agree with toofar's feedback. Some 
general points though.

It might not be clear that this saves the current top-level domain, although I 
don't think there's much we can do about that.

It might be nice to strip the scheme and maybe allow child domains as well, 
although that might be even more confusing (I always prefer *.foo.bar rules 
over foo.bar rules).

This will probably make the performance issues for per-domain settings much 
more apparent for normal users since they are likely to hit these buttons by 
accident (or on purpose for many sites). I don't think we can do much about 
that other than fixing it directly.

**Comment by The-Compiler** [comment]
I noticed this does the opposite of what you'd expect for the ssl_strict setting - I think I'm just going to remove it for that setting until it's renamed (with the values reversed), IIRC there was an issue for that.

**Comment by The-Compiler** [comment]
This should be finished now (if CI passes) from my perspective.

> It might not be clear that this saves the current top-level domain, although I
> don't think there's much we can do about that.
> 
> It might be nice to strip the scheme and maybe allow child domains as well,
> although that might be even more confusing (I always prefer *.foo.bar rules
> over foo.bar rules).

I first thought about that, but decided to leave it out - I'd rather have the user need to answer another confirmation (i.e. press Y/N again) than giving out more permissions than they intended to.

> This will probably make the performance issues for per-domain settings much
> more apparent for normal users since they are likely to hit these buttons by
> accident (or on purpose for many sites). I don't think we can do much about
> that other than fixing it directly.

You could argue the same about e.g. the `tSh` bindings. Also if I'm not mistaken, the only situation where many of those settings would make a difference is when a website requests a permission (i.e., before the dialog is shown).

**Comment by The-Compiler** [comment]
I noticed QtWebKit included path/query/fragment/etc. in URLs shown in permission requests - so I stripped them (and trailing slashes on QtWebEngine) in 392bb2fdec44b559ae1587cb5f236c0de92151f7.

File: tests/end2end/features/prompts.feature [hunk 2]
```diff
@@ -263,6 +272,17 @@ Feature: Prompts
         And I run :prompt-accept no
         Then the javascript message "notification permission denied" should be logged
 
+    @qtwebengine_notifications
+    Scenario: notifications with ask -> false and save
+        Given I have a fresh instance
```
Comment (toofar) [inline tests/end2end/features/prompts.feature]: I don't know how much you care about speed with these things (or how slow starting a fresh instance is) but would it be faster to just set the setting back to default afterwards?

Comment (The-Compiler) [reply tests/end2end/features/prompts.feature]: That'd be great, but unfortunately QtWebEngine somehow remembers those permissions (until the next restart) as well.

**Pull request merged**